monitor/service.go

```diff
@@ -156,52 +172,17 @@ func (m *Monitor) Statistics(tags map[string]string) ([]*Statistic, error) {
 			statistic.Tags[k] = v
 		}
 
-		// Every other top-level expvar value is a map.
-		m := kv.Value.(*expvar.Map)
+		for k, v := range s.Tags() {
+			statistic.Tags[k] = v
+		}
 
-		m.Do(func(subKV expvar.KeyValue) {
-			switch subKV.Key {
-			case "name":
-				// straight to string name.
-				u, err := strconv.Unquote(subKV.Value.String())
-				if err != nil {
-					return
-				}
-				statistic.Name = u
-			case "tags":
-				// string-string tags map.
-				n := subKV.Value.(*expvar.Map)
-				n.Do(func(t expvar.KeyValue) {
-					u, err := strconv.Unquote(t.Value.String())
-					if err != nil {
-						return
-					}
-					statistic.Tags[t.Key] = u
-				})
-			case "values":
-				// string-interface map.
-				n := subKV.Value.(*expvar.Map)
-				n.Do(func(kv expvar.KeyValue) {
-					var f interface{}
-					var err error
-					switch v := kv.Value.(type) {
-					case *expvar.Float:
-						f, err = strconv.ParseFloat(v.String(), 64)
-						if err != nil {
-							return
-						}
-					case *expvar.Int:
-						f, err = strconv.ParseInt(v.String(), 10, 64)
-						if err != nil {
-							return
-						}
-					default:
-						return
-					}
-					statistic.Values[kv.Key] = f
-				})
-			}
-		})
+		if s.Refs() == 1 {
+			// In this case, the monitor's reference is the only remaining
+			// open reference to the Statistics object, s. Since there are no other open
+			// references (and hence no more updates possible), it is safe for the monitor
+			// to release its reference, so it does so now.
+			s.Close()
+		}
```

Comment:
The uncool thing about this test is that it assumes that there is only ever one observer who adds references to Statistics by calling Open. If that isn't true, this test doesn't make sense.

Now, it happens, that in the current runtime, there will only be one other caller to Statistics.Open(), that being monitor, and it seems likely that this will always be the case. However, if this assumption every changed, this code would result in a Statistics leak. 

I have ideas about how to fix this issue with the Go type system, but they seem like overkill at this point.